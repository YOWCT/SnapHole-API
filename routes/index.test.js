let request = require('request-promise-native')
let _ = require('lodash')

function sanitize (data, keys) {
  return keys.reduce((result, key) => {
    const val = _.get(result, key)
    if (!val || _.isArray(val) || _.isObject(val)) {
      return result
    } else {
      return _.set(_.cloneDeep(result), key, '[SANITIZED]')
    }
  }, data)
}

test('router sr_information adds information', async () => {
  let response
  try {
    let formObject = {
      clientInformation: 'n/a',
      imgName: '123456.jpeg',
      uuid: '123456',
      long: '-76.0804324',
      lat: '45.249814'
    }
    response = await request.post({
      url: `${global.__DOMAIN__}/sr_information`,
      form: formObject
    })
    response = sanitize(JSON.parse(response), ['_id', 'timestamp'])
    expect(response).toMatchSnapshot()
  } catch (e) {
    console.log(`err ${e}`)
  }
})
