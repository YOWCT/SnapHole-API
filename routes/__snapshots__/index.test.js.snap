// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router sr_information adds information 1`] = `
Object {
  "__v": 0,
  "_id": "[SANITIZED]",
  "clientInformation": "N/A",
  "fkPhid": "123456",
  "imgName": "123456.jpeg",
  "imgUrl": "https://s3.amazonaws.com/snaphole/123456.jpeg",
  "imgurUrl": "https://s3.amazonaws.com/snaphole/123456.jpeg",
  "location": Object {
    "coordinates": Array [
      -76.0804324,
      45.249814,
    ],
    "type": "Point",
  },
  "timestamp": "[SANITIZED]",
}
`;
